src/Tizen.MachineLearning.Service/Tizen.MachineLearning.Service/Service.cs

```diff
@@ -0,0 +1,696 @@
+
+            IntPtr value = IntPtr.Zero;
+            MLServiceError ret = Interop.Service.GetInformation(_handle, name, out value);
+            MLService.CheckException(ret, "Failed to get service information");
+
+            return Interop.Util.IntPtrToString(value);
+        }
+
+        /// <summary>
+        /// Adds an input data to process the model in service instance.
+        /// </summary>
+        /// <param name="data">The data to be processed.</param>
+        /// <param name="name">The name of node in the pipeline. You can set NULL if serivce is constructed from model configuration.</param>
+        /// <feature>http://tizen.org/feature/machine_learning.service</feature>
+        /// <exception cref="ArgumentException">Thrown when the method failed due to an invalid parameter.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed to stop the service.</exception>
+        /// <since_tizen> 12 </since_tizen>
+        public void Request(TensorsData data, string name = "")
+        {
+            if (data == null)
+                throw MLServiceExceptionFactory.CreateException(MLServiceError.InvalidParameter, "Given data is invalid");
+
+            IntPtr nameHandle = Interop.Util.StringToIntPtr(name);
+            MLServiceError ret = Interop.Service.Request(_handle, nameHandle, data.GetHandle());
+            MLService.CheckException(ret, "Failed to request service");
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object.
+        /// </summary>
+        /// <since_tizen> 12 </since_tizen>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Releases any unmanaged resources used by this object including opened handle.
+        /// </summary>
+        /// <param name="disposing">If true, disposes any disposable objects. If false, does not dispose disposable objects.</param>
+        /// <since_tizen> 12 </since_tizen>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+                return;
+
+            if (disposing)
+            {
+                // release managed object
+            }
+
+            // release unmanaged objects
+            if (_handle != IntPtr.Zero)
+            {
+                MLServiceError ret = Interop.Service.Destroy(_handle);
+                if (ret != MLServiceError.None)
+                    Log.Error(MLService.TAG, "Failed to destroy the service handle");
+
+                _handle = IntPtr.Zero;
+            }
+
+            _disposed = true;
+        }
+
+        /// <summary>
+        /// The Query class provides service based on MlInformation.
```

Comment:
The Query class provides machine learning service for edge connection.

let's consider using key-word edge or connection to remote device.

Reply:
Updated! Thank you